components/core/src/os/filesystem/windows.rs

```diff
@@ -17,7 +17,7 @@ use std::ffi::CStr;
 use std::path::Path;
 use std::io;
 
-pub fn chown(r_path: *const c_char, uid: u32, gid: u32) -> c_int {
+pub fn chown(r_path: *const c_char, uid: String, gid: String) -> c_int {
```

Comment:
Changing this signature is going to be a problematic unless you add platform awareness around https://github.com/habitat-sh/habitat/blob/master/components/core/src/util/perm.rs#L65 

Perhaps, since users are represented so differently, we expose a User struct whose internals vary by platform (one implemented in users/windows and one in users/linux.  Then the chown and chmod signatures could take the users::User struct and each look for what they care about.